src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.OneNote/Main.cs

```diff
@@ -83,28 +92,52 @@ public void Init(PluginInitContext context)
         /// <returns>A filtered list, can be empty when nothing was found</returns>
         public List<Result> Query(Query query)
         {
-            if (!_oneNoteInstalled || query is null || string.IsNullOrWhiteSpace(query.Search))
+            if (!_oneNoteInstalled || query is null || string.IsNullOrWhiteSpace(query.Search) || cache is null)
             {
                 return new List<Result>(0);
             }
 
-            var pages = OneNoteProvider.FindPages(query.Search);
+            // If there's cached results for this query, return immediately, otherwise wait for delayedExecution.
+            var results = cache.Get<List<Result>>(query.Search);
```

Comment:
if note from cache is selected, but it was permanently deleted, launcher is crashing.. Is there a way to check whether cached notes still exist?